functions/get-order.ts

```diff
@@ -22,7 +23,12 @@ export async function onRequest(ctx: Context): Promise<Response> {
     if (!reloadlyTransaction) {
       return Response.json("Order not found.", { status: 404 });
     } else if (reloadlyTransaction.status && reloadlyTransaction.status == "SUCCESSFUL") {
-      return Response.json(reloadlyTransaction, { status: 200 });
+      try {
+        const product = await getGiftCardById(reloadlyTransaction.product.productId, accessToken);
+        return Response.json({ transaction: reloadlyTransaction, product: product }, { status: 200 });
+      } catch (error) {
+        return Response.json({ transaction: reloadlyTransaction, product: null }, { status: 200 });
```

Comment:
Is seems strange to return a `200` status if we have no product to return but the `reloadlyTransaction` was successful, why do we do this?

Reply:
Yes, I also found it strange but it was a problem on the Reloadly side. When a product is out of stock or unavailable for any other reason, they stop returning the product. You get a 404 or 400 if you try to fetch it.  

Specific to this code line, it is possible that someone purchased a virtual card in the past but now that product is out of stock or unavailable for some other reason. In this case, the user must see at least their redeem code. 

Most of the time the product will be available. In this case, the user who purchased a virtual card in the past should see the redeem code and complete product details. 

Our response of 200 is okay here because even though we don't have all the product details, we have the most important details (redeem code).

Reply:
That makes sense I guess. And in this rare scenario where the product is null but we have a redeem code at that point it's the responsibility of the card provider? or are we technically still in control i.e can we revert everything or is it the case the redeem code once minted there's no going back?

As you say rare, but in the case it happens and we give a redeem code with no product, would the user still receive everything as normal and it would be a valid card, yeah? I'd have thought no product means there is nothing to redeem so even with a redeem code, what product are they redeeming sort of thing.

Reply:
> is it the case the redeem code once minted there's no going back

This is true. We don't have control once we get the redeem code. Even Reloadly doesn't have much control and they can't refund or replace.  That is why, we have disclaimers everywhere.

> As you say rare, but in the case it happens and we give a redeem code with no product, would the user still receive everything as normal 

Correct. 

> I'd have thought no product means there is nothing to redeem so even with a redeem code

There are product details with the Transaction object. I fetched full product for some extra properties, which technically should have been part of the transaction I think but Reloadly didn't do so. For example product image link. The product object has it, transaction doesn't. Even though the transaction object has other properties like product name, price, country... etc. You can look at the object types if you want to learn more.